cats/cats-sql/src/main/kotlin/com/netflix/spinnaker/cats/sql/cache/SqlCache.kt

```diff
@@ -78,38 +80,14 @@ class SqlCache(
   }
 
   /**
-   * Only evicts cache records but not relationship rows
+   * Evicts cache records, but does not evict relationship rows.
+   *
+   * @param type the type of the records that will be removed.
+   * @param ids the ids of the records that will be removed.
    */
   override fun evictAll(type: String, ids: Collection<String>) {
-    if (ids.isEmpty()) {
-      return
-    }
-
-    log.info("evicting ${ids.size} $type records")
-
-    var deletedCount = 0
-    var opCount = 0
-    try {
-      ids.chunked(dynamicConfigService.getConfig(Int::class.java, "sql.cache.read-batch-size", 500)) { chunk ->
-        withRetry(RetryCategory.WRITE) {
-          jooq.deleteFrom(table(sqlNames.resourceTableName(type)))
-            .where(field("id").`in`(*chunk.toTypedArray()))
-            .execute()
-        }
-        deletedCount += chunk.size
-        opCount += 1
-      }
-    } catch (e: Exception) {
-      log.error("error evicting records", e)
-    }
-
-    cacheMetrics.evict(
-      prefix = name,
-      type = type,
-      itemCount = ids.size,
-      itemsDeleted = deletedCount,
-      deleteOperations = opCount
-    )
+    val hashedIds = ids.asSequence().map { getIdHash(it) }.toSet()
```

Comment:
_Potentially dumb kotlin question alert:_ What's the purpose of the `asSequence` and `toSet` calls here if `evictAllInternal` just accepts `Collection<String>`?

Reply:
Oh I'm going to just resolve this and my other related comment as I see @ezimanyi already asked the same question! (Sorry, I was halfway done reviewing this when he posted his review and didn't bother looking at his comments first...)